pkgs/top-level/all-packages.nix

```diff
@@ -1790,7 +1790,7 @@ in
 
   eschalot = callPackage ../tools/security/eschalot { };
 
-  esphome = callPackage ../tools/misc/esphome { };
+  esphome = python3Packages.callPackage ../tools/misc/esphome { };
```

Comment:
You shouldn't use `python3Packages.callPackages` because that makes it impossible to override the `python` argument.

Reply:
There is no need to override python argument after my change.

Reply:
Will move the change to other PR